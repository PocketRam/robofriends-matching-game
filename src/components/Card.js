import React from 'react';
import PropTypes from 'prop-types';

const Card = ({ id, alt, flippedCard, matchingCards, openedCard, onCardClick }) => {
  const cardClasses = 'card tc grow bg-light-green br3 pa2 ma2 dib bw2 shadow-5';
  const isMatchingCard = matchingCards.find(card => card[0] === id);
  const shouldCardBeOpen = isMatchingCard || flippedCard[0] === id  || openedCard[0] === id;

  return (
    <li className={`${cardClasses} ${shouldCardBeOpen ? 'open': 'closed'}`}
        onClick={onCardClick}>
      <img id={id} alt={alt} src={`https://robohash.org/${alt}?size=100x100`} />
    </li>
  );
}

Card.propTypes = {
  id: PropTypes.string.isRequired,
  alt: PropTypes.string.isRequired,
  flippedCard: PropTypes.array.isRequired,
  matchingCards: PropTypes.array.isRequired,
  openedCard: PropTypes.array.isRequired,
  onCardClick: PropTypes.func.isRequired
};

export default Card;