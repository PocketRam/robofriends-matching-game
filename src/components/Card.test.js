import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

it('renders without crashing', () => {
  expect(shallow(<Card id={1}/>)).toMatchSnapshot();
});