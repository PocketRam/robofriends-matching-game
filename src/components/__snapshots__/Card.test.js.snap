// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    alt="robot1"
    flippedCard={Array []}
    id="robot1-1"
    matchingCards={Array []}
    onCardClick={[Function]}
    openedCard={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <img
        alt="robot1"
        id="robot1-1"
        src="https://robohash.org/robot1?size=100x100"
/>,
      "className": "card tc grow bg-light-green br3 pa2 ma2 dib bw2 shadow-5 closed",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "robot1",
        "id": "robot1-1",
        "src": "https://robohash.org/robot1?size=100x100",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          alt="robot1"
          id="robot1-1"
          src="https://robohash.org/robot1?size=100x100"
/>,
        "className": "card tc grow bg-light-green br3 pa2 ma2 dib bw2 shadow-5 closed",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "robot1",
          "id": "robot1-1",
          "src": "https://robohash.org/robot1?size=100x100",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
